src/R/Core/Impl/Formatting/RFormatter.cs

```diff
@@ -440,7 +440,10 @@ public class RFormatter {
         private void AppendOperator() {
             string text = _textProvider.GetText(_tokens.CurrentToken);
             if (_tokens.Position == 0 || TokenOperator.IsUnaryOperator(_tokens, _textProvider, TokenOperator.GetOperatorType(text), -1)) {
-                AppendToken(leadingSpace: true, trailingSpace: false);
+                // Determine if we need to insert leading space. We do in 'x + -1' but not in x + (-y + 1)
+                var precedingTokenType = _tokens.LookAhead(-1).TokenType;
+                bool leadingSpace = precedingTokenType == RTokenType.Operator;
```

Comment:
How does this handle cases where the preceding operator is also unary? E.g. `+-x` or `~-x`.

Also, for `x + -1`, wouldn't both spaces just come from the `+`?